src/main/java/bot/ControlFlow.java

```diff
@@ -0,0 +1,133 @@
+package bot;
+
+import bot.command.Command;
+import bot.command.DeadlineCommand;
+import bot.command.DeleteCommand;
+import bot.command.EventCommand;
+import bot.command.ListCommand;
+import bot.command.MarkCommand;
+import bot.command.TerminateCommand;
+import bot.command.TodoCommand;
+import bot.command.UnmarkCommand;
+import bot.command.FindCommand;
+
+import bot.exception.DateTimeParseBotException;
+import bot.exception.FileErrorBotException;
+import bot.exception.IllegalExpressionBotException;
+import bot.exception.IncompleteBotException;
+
+import bot.parsers.InputParser;
+import bot.task.TaskList;
+
+
+public class ControlFlow {
+
+    private static final String MARK = "mark";
+    private static final String UNMARK = "unmark";
+    private static final String LIST = "list";
+    private static final String TODO = "todo";
+    private static final String EVENT = "event";
+    private static final String DEADLINE = "deadline";
+    private static final String TERMINATE = "bye";
+    private static final String DELETE = "delete";
+    private static final String FIND = "find";
+
+    private final TaskList taskList;
+
+    public ControlFlow(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Returns the Command object that is to be executed based on user input.
+     *
+     * @param str the input by the user
+     * @return Command the necessary tasks that needed to be executed as per the user's input
+     * @throws IllegalExpressionBotException
+     * @throws IncompleteBotException
+     * @throws FileErrorBotException
+     * @throws DateTimeParseBotException
+     */
+    public Command execute(String str) throws IllegalExpressionBotException,
+            IncompleteBotException, FileErrorBotException, DateTimeParseBotException {
+        if (str.isBlank()) {
+            throw new IncompleteBotException("OOPS!!! There is no bot.task specified.");
+        }
+        String[] strSplit = InputParser.getSplitAtSpace(str);
+        String taskName = InputParser.getLeftOfSplit(strSplit);
+        String remainder = "";
+        if (strSplit.length > 1) {
+            remainder = InputParser.getRightOfSplit(strSplit);
+        }
+        String[] taskSplit;
+        String taskDetail;
+        String timeline;
+        String[] timelineArr;
+        String timeFrom;
+        String timeTo;
```

Comment:
Since the method is so long, you might want to utilise blank spaces between rows to structure the segments of code while making the code easier to read.